unittests/protocol_feature_tests.cpp

```diff
@@ -526,182 +526,6 @@ BOOST_AUTO_TEST_CASE( replace_deferred_test ) try {
 
 } FC_LOG_AND_RETHROW()
 
-BOOST_AUTO_TEST_CASE( no_duplicate_deferred_id_test ) try {
-   tester c( setup_policy::preactivate_feature_and_new_bios );
-   tester c2( setup_policy::none );
-
-   c.preactivate_builtin_protocol_features( {builtin_protocol_feature_t::crypto_primitives} );
-   c.produce_block();
-   c.create_accounts( {"alice"_n, "test"_n} );
-   c.set_code( "test"_n, test_contracts::deferred_test_wasm() );
-   c.set_abi( "test"_n, test_contracts::deferred_test_abi() );
-   c.produce_block();
-
-   push_blocks( c, c2 );
-
-   c2.push_action( "test"_n, "defercall"_n, "alice"_n, fc::mutable_variant_object()
-      ("payer", "alice")
-      ("sender_id", 1)
-      ("contract", "test")
-      ("payload", 50)
-   );
-
-   c2.finish_block();
-
-   BOOST_CHECK_EXCEPTION(
-      c2.produce_block(),
-      fc::exception,
-      fc_exception_message_is( "no transaction extensions supported yet for deferred transactions" )
-   );
-
-   c2.produce_empty_block( fc::minutes(10) );
-
-   transaction_trace_ptr trace0;
-   auto h2 = c2.control->applied_transaction.connect( [&](std::tuple<const transaction_trace_ptr&, const packed_transaction_ptr&> x) {
-      auto& t = std::get<0>(x);
-      if( t && t->receipt && t->receipt->status == transaction_receipt::expired) {
-         trace0 = t;
-      }
-   } );
-
-   c2.produce_block();
-
-   h2.disconnect();
-
-   BOOST_REQUIRE( trace0 );
-
-   c.produce_block();
-
-   const auto& index = c.control->db().get_index<generated_transaction_multi_index,by_trx_id>();
-
-   transaction_trace_ptr trace1;
-   auto h = c.control->applied_transaction.connect( [&](std::tuple<const transaction_trace_ptr&, const packed_transaction_ptr&> x) {
-      auto& t = std::get<0>(x);
-      if( t && t->receipt && t->receipt->status == transaction_receipt::executed) {
-         trace1 = t;
-      }
-   } );
-
-   BOOST_REQUIRE_EQUAL(0u, index.size());
-
-   c.push_action( config::system_account_name, "reqauth"_n, "alice"_n, fc::mutable_variant_object()
-      ("from", "alice"),
-      5, 2
-   );
```

Comment:
I think most of this test case is still salvageable even if this action can no longer be pushed.

Maybe I'm not remembering the details of this test correctly, but I don't think pushing this delayed transaction really even gets us much in this test. Maybe it was just there to ensure that a deferred transaction generated prior to activation of the protocol feature would still be retired correctly.

So I think you should be able to remove it, adapt some of the `BOOST_REQUIRE_EQUAL`s below, and still keep the bulk of this test case which is checking the transaction extensions which are there to allow for unique deferred transaction IDs.

Reply:
Thanks. I have restored and adapted `protocol_feature_tests/no_duplicate_deferred_id_test` instead of removing it outright.